### ISSUE TITLE
Layout. "Invariant Violation: processUpdates()" error when layout's child presence is changed

### ISSUE THREAD

**Issue by samoshkin** [tleunen/react-mdl#280]
I'm using `react-mdl@1.5.2`, and patched version of `material.js` and `material.css` according to [React-MDL requirements](https://github.com/tleunen/react-mdl#requirements), not `material-design-lite` counter parts, because I'm aware of [issue #1356](https://github.com/google/material-design-lite/issues/1356).

Layout component has two children: `Header` and `div.footer`. Initially `div.footer` is visible and rendered as direct child of Layout component. Then, when state `isFooterVisible` changes, div.footer will not be rendered at all during next cycle.

```
renderFooter(){
  if (!this.state.isFooterVisible) return null;

  return (
    <div className="Layout-footer">
      <Button onClick={this.toggleFooter}>Hide Footer</Button>
    </div>
  );
}

render(){
  return (
    <Layout fixedHeader>
      {this.renderHeader()}
      {this.renderFooter()}
    </Layout>
  );
}
```

Given that, when `isFooterVisible` becomes false, React fails at reconcile phase with following error.

> Uncaught Invariant Violation: processUpdates(): Unable to find child 1 of element. This probably means the DOM was unexpectedly mutated (e.g., by the browser), usually due to forgetting a &lt;tbody&gt; when using tables, nesting tags like &lt;form&gt;, &lt;p&gt;, or &lt;a&gt;, or using non-SVG elements in an &lt;svg&gt; parent. Try inspecting the child nodes of the element with React ID `.0`

I created example repo, so you can reproduce the error. See https://github.com/samoshkin/react-mdl-layout-reconcile-issue

**Comment by tleunen** [comment]
Thanks @samoshkin for the great bug report! I'll use your repo to help me find the root cause to this.

**Comment by tleunen** [comment]
Thanks a lot @samoshkin for the bug repo. It helped me a lot having quickly a test example and work a fix on it.
I moved the button in the layout so I could easily toggle the footer.
![mdl](https://cloud.githubusercontent.com/assets/1972567/14407778/ec766fa4-fea2-11e5-8aa7-da9c091f1e7d.gif)

A fix will follow soon :)
The fix is mostly inside the patched version of MDL. Instead of letting him creating a dom element for the  layout inner container, it's created by React. Thus it knows where the node is.

**Comment by samoshkin** [comment]
Wow, thanks a lot for fixing this. Looking forward for the next release :)

**Comment by tleunen** [comment]
Released as 1.5.3 :)
